### ISSUE TITLE
FID increases for larger training datasets

### ISSUE THREAD

**Issue by slala2121** [mit-han-lab/data-efficient-gans#55]
I trained a generator using 3.8K images on 256x256 following the settings described and trained another generator using 750 images with the same settings but with regularization increased from 1 to 10 as in the paper. 

While the training FID-5K score is lower when training on 3.8K vs 750 (both training FID curves are stable), I find that the the FID score computed using the generator trained on 3.8K gives much higher FID scores compared to that attained under 750. 

I am using the same code to compute the FID scores so it is unclear to me why this is the case. Have you encountered this issue?

**Comment by zsyzzsoft** [comment]
"While the training FID-5K score is lower when training on 3.8K vs 750" What does this sentence mean? Could you give more details?

**Comment by slala2121** [comment]
When training on 3.8K images, the fid-5k score monitored through training
converges to around 8.0. When training on 750 images, the fid-5k score
monitored through training converges to around 33.

When I evaluate the fid-50k score for the generative model trained on 750
images, it has an fid-50k score around 21. When I evaluate the fid-50k
score the generative model trained on 3.8K images, it has an fid-50k score
around 50.

Best,

Sayeri Lala
PhD candidate | Electrical Engineering | Princeton University

**Comment by zsyzzsoft** [comment]
This is probably caused by the inconsistency between training and testing. The FID-50k score of a model should be smaller than FID-5k of the same model.

**Comment by slala2121** [comment]
By inconsistency, do you mean with respect to image preprocessing?

Best,

Sayeri Lala
PhD candidate | Electrical Engineering | Princeton University

**Comment by slala2121** [comment]
Or are there other differences bet training and testing for the generator
e.g., dropout, batch norm? Thanks.

Best,

Sayeri Lala
PhD candidate | Electrical Engineering | Princeton University

**Comment by zsyzzsoft** [comment]
Most likely to be caused by image preprocessing.

**Comment by zsyzzsoft** [comment]
Another guess - maybe you computed the FID statistics of real images with 750 images, but when you evaluate the model trained with 3.8k images, it still uses the cached statictics with 750 images. Maybe you can clear the cached files or rename the dataset, and then re-evaluate the model trained with 3.8k images.

**Comment by slala2121** [comment]
Thanks, I'll look into it.

Best,

Sayeri Lala
PhD candidate | Electrical Engineering | Princeton University

**Comment by slala2121** [comment]
I made some fixes so now the results match. Thanks for your thoughts!
